kubernetes.go

```diff
@@ -548,6 +558,53 @@ func (svc *KubernetesServiceOp) Delete(ctx context.Context, clusterID string) (*
 	return resp, nil
 }
 
+// DeleteSelective deletes a Kubernetes cluster and the specified associated resources.
+// Users can choose to delete specific volumes, volume snapshots or load balancers along with the cluster
+// There is no way to recover a cluster or the specified resources once destroyed.
+func (svc *KubernetesServiceOp) DeleteSelective(ctx context.Context, clusterID string, request *KubernetesClusterDeleteSelectiveRequest) (*Response, error) {
+	path := fmt.Sprintf("%s/%s/destroy_with_associated_resources/selective", kubernetesClustersPath, clusterID)
+	req, err := svc.client.NewRequest(ctx, http.MethodDelete, path, request)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := svc.client.Do(ctx, req, nil)
+	if err != nil {
+		return resp, err
+	}
+	return resp, nil
+}
+
+// DeleteDangerous deletes a Kubernetes cluster and all its associated resources. There is no way to recover a cluster
+// or it's associated resources once destroyed.
+func (svc *KubernetesServiceOp) DeleteDangerous(ctx context.Context, clusterID string) (*Response, error) {
+	path := fmt.Sprintf("%s/%s/destroy_with_associated_resources/dangerous", kubernetesClustersPath, clusterID)
+	req, err := svc.client.NewRequest(ctx, http.MethodDelete, path, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := svc.client.Do(ctx, req, nil)
+	if err != nil {
+		return resp, err
+	}
+	return resp, nil
+}
+
+// ListAssociatedResourcesForDeletion lists a Kubernetes cluster's resources that can be selected
+// for deletion along with the cluster. See DeleteSelective
+// Associated resources include volumes, volume snapshots and load balancers.
+func (svc *KubernetesServiceOp) ListAssociatedResourcesForDeletion(ctx context.Context, clusterID string) (*Response, error) {
```

Comment:
Do we want to unmarshal this response into a struct?

Reply:
@andrewsomething - done. Thank you!